### PR TITLE
Signin page

### PR DESCRIPTION
**Pull request by radiansz**

### CODE REVIEW

File: client/routes/signin/components/SigninContent.css [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  & .signin-content__language-panel {
+    background-color: #fff;
+    border-radius: 4px;
+    border: 1px solid color(white blackness(11,4%));
+    margin: 10px auto 15px;
+    padding: 20px 50px;
+    width: 472px;
+  }
+
+  & .ReactTabs__TabList {
```
Comment (radiansz) [inline client/routes/signin/components/SigninContent.css]: I dunno should i create wrapper tabs component, make or global styles or keep it here

File: client/components/SigninHeader/SigninHeader.css [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  justify-content: center;
+}
+
+.signin-header__container {
+  display: flex;
+  max-width: 1150px;
+  min-width: 960px;
+}
+
+.signin-header__nav {
```
Comment (zouxuoz) [inline client/components/SigninHeader/SigninHeader.css]: @radiansz never used?

File: client/components/SigninHeader/SigninHeader.css [hunk 3]
```diff
@@ -0,0 +1,20 @@
+@import "styles/colors.css";
+
+.signin-header {
+  background-color: var(--darkblueColor);
+  display: flex;
+  height: 65px;
```
Comment (zouxuoz) [inline client/components/SigninHeader/SigninHeader.css]: @radiansz i think child must expand parent, not the opposite

File: client/components/SigninHeader/SigninHeader.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import SigninHeaderLogo from './SigninHeaderLogo';
+import './SigninHeader.css';
+
+const SigninHeader = () => (
+  <header className="signin-header">
```
Comment (zouxuoz) [inline client/components/SigninHeader/SigninHeader.js]: must use `lib/bem`

File: client/components/SigninHeader/SigninHeader.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    <div className="signin-header__container">
```
Comment (zouxuoz) [inline client/components/SigninHeader/SigninHeader.js]: must use `lib/bem`

File: client/components/SigninHeader/SigninHeaderLogo.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import logo from 'assets/logo.svg';
+import './SigninHeaderLogo.css';
+
+const SigninHeaderLogo = () => (
+  <div className="signin-header__logo" dangerouslySetInnerHTML={{ __html: logo }} />
```
Comment (zouxuoz) [inline client/components/SigninHeader/SigninHeaderLogo.js]: must use `lib/bem`

File: client/components/ui/Button/Button.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+const Button = ({ classModifiers, children, onClick, type = 'button', loading }) => {
+  return (
+    <button {...classes(':root', { ...classModifiers, loading: loading })} type={type} onClick={onClick}>{children}</button>
+  )
+};
+
+Button.propTypes = {
+  children: PropTypes.node.isRequired,
+  loading: PropTypes.bool,
```
Comment (zouxuoz) [inline client/components/ui/Button/Button.js]: @radiansz why `loading` not in `classModifiers`?

File: client/components/ui/Input/Input.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+import './Input.css';
+
+const classes = bem('input');
+
+const Input = ({ label, type, meta, input, placeholder, classModifiers }) => {
+  const hasError = !!meta.error;
+
+  return (
+    <div {...classes(':root', { ...classModifiers, invalid: hasError })}>
+      {label && <label htmlFor={input.name}>{label}</label>}
```
Comment (zouxuoz) [inline client/components/ui/Input/Input.js]: @radiansz maybe use `if/else` or transfer to a separate function?

File: client/components/ui/Input/Input.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+      <input {...input} type={type} placeholder={placeholder} />
+      {hasError && <span className="input__error">{meta.error}</span>}
```
Comment (zouxuoz) [inline client/components/ui/Input/Input.js]: @radiansz maybe use if/else or transfer to a separate function?

File: client/layouts/SigninLayout/SigninLayout.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import SigninHeader from 'components/SigninHeader';
+import './SigninLayout.css';
+
+const signinLayout = (ComposedComponent) => {
+  function SigninLayout(props) {
+    return (
+      <main className="signin-layout">
```
Comment (zouxuoz) [inline client/layouts/SigninLayout/SigninLayout.js]: must use `lib/bem`

File: client/styles/select.css [hunk 11]
```diff
@@ -0,0 +1,382 @@
+/**
```
Comment (zouxuoz) [inline client/styles/select.css]: i think better use react-select variables for customize styles (/summon @rosendi)

File: client/styles/tabs.css [hunk 12]
```diff
@@ -0,0 +1,18 @@
+.ReactTabs__TabList {
```
Comment (zouxuoz) [inline client/styles/tabs.css]: @radiansz, @rosendi don't like not our styles in our styles, what we can do with that?

File: client/routes/signin/components/SigninContent.js [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+  agses: {
+    id: 'signin-content__agses',
+    defaultMessage: 'Agses-карта',
+  },
+});
+
+const SigninContent = () => (
+  <div className="signin-content">
+    <div className="signin-content__tabs-panel">
+      <Tabs className="hello-world">
```
Comment (zouxuoz) [inline client/routes/signin/components/SigninContent.js]: wtf? 😈

File: client/routes/signin/components/SigninContent.js [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+  },
+
+  agses: {
+    id: 'signin-content__agses',
+    defaultMessage: 'Agses-карта',
+  },
+});
+
+const SigninContent = () => (
+  <div className="signin-content">
```
Comment (zouxuoz) [inline client/routes/signin/components/SigninContent.js]: must use `lib/bem`

File: client/routes/signin/components/SigninContent.js [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+    <div className="signin-content__tabs-panel">
```
Comment (zouxuoz) [inline client/routes/signin/components/SigninContent.js]: must use `lib/bem`

File: client/routes/signin/components/SigninContent.js [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+      <Tabs className="hello-world">
+        <TabList>
+          <Tab>
+            <div>
+              <FormattedMessage {...messages.ordinary} />
+            </div>
+          </Tab>
+
+          <Tab className="signin-content__disabled-tab">
```
Comment (zouxuoz) [inline client/routes/signin/components/SigninContent.js]: it must be tab modifier

File: client/routes/signin/components/SigninContent.js [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+            <div>
+              <FormattedMessage {...messages.corporate} />
+            </div>
+          </Tab>
+
+          <Tab className="signin-content__disabled-tab">
```
Comment (zouxuoz) [inline client/routes/signin/components/SigninContent.js]: it must be tab modifier

File: client/routes/signin/components/SigninForm.js [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+    id: 'signin-content__agses',
+    defaultMessage: 'Войти',
+  },
+});
+
+const SigninForm = ({ handleSubmit, error, intl, submitting }) => (
+  <div className="signin-form">
+    <form onSubmit={handleSubmit}>
+
+      <div className="signin-form__field">
```
Comment (zouxuoz) [inline client/routes/signin/components/SigninForm.js]: must use `lib/bem`

File: client/routes/signin/components/SigninForm.js [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+  },
+
+  signin: {
+    id: 'signin-content__agses',
+    defaultMessage: 'Войти',
+  },
+});
+
+const SigninForm = ({ handleSubmit, error, intl, submitting }) => (
+  <div className="signin-form">
```
Comment (zouxuoz) [inline client/routes/signin/components/SigninForm.js]: must use `lib/bem`

File: client/routes/signin/components/SigninForm.js [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+    <form onSubmit={handleSubmit}>
+
+      <div className="signin-form__field">
+        <Field name="login" component={Input} type="text" placeholder={intl.formatMessage(messages.loginPlaceholder)} />
+      </div>
+
+      <div className="signin-form__field">
```
Comment (zouxuoz) [inline client/routes/signin/components/SigninForm.js]: must use `lib/bem`

File: client/routes/signin/components/SigninForm.js [hunk 21]
```diff
@@ -0,0 +1,54 @@
...
+        <Field error={error} name="password" component={Input} type="password" placeholder={intl.formatMessage(messages.passwordPlaceholder)} />
+      </div>
+
+      <div className="signin-form__footer">
```
Comment (zouxuoz) [inline client/routes/signin/components/SigninForm.js]: must use `lib/bem`

File: client/routes/signin/modules/signin/sagas.js [hunk 22]
```diff
@@ -59,12 +59,12 @@ function* signin() {
       let error = 'Unknown error';
 
       if (err.errors[0].explain === 'wrong_credentials') {
-        error = 'Wrong password';
+        error = 'Проверьте правильность введеных данных';
```
Comment (zouxuoz) [inline client/routes/signin/modules/signin/sagas.js]: i18n?

File: client/components/ui/Button/Button.js (hunk 7 above)
Comment (radiansz) [inline client/components/ui/Button/Button.js]: Such thing was proposed by @rosendi

File: client/components/ui/Input/Input.js (hunk 8 above)
Comment (radiansz) [inline client/components/ui/Input/Input.js]: fixed

File: client/routes/signin/components/SigninContent.js (hunk 14 above)
Comment (radiansz) [inline client/routes/signin/components/SigninContent.js]: fixed

File: client/routes/signin/components/SigninContent.js (hunk 15 above)
Comment (radiansz) [inline client/routes/signin/components/SigninContent.js]: fixed

File: client/routes/signin/components/SigninForm.js (hunk 18 above)
Comment (radiansz) [inline client/routes/signin/components/SigninForm.js]: fixed

File: client/routes/signin/components/SigninForm.js (hunk 19 above)
Comment (radiansz) [inline client/routes/signin/components/SigninForm.js]: fixed

File: client/routes/signin/components/SigninForm.js (hunk 21 above)
Comment (radiansz) [inline client/routes/signin/components/SigninForm.js]: fixed

File: client/routes/signin/components/SigninForm.js (hunk 20 above)
Comment (radiansz) [inline client/routes/signin/components/SigninForm.js]: fixed

File: client/layouts/SigninLayout/SigninLayout.js (hunk 10 above)
Comment (radiansz) [inline client/layouts/SigninLayout/SigninLayout.js]: Probably wont be here after merge of #292 in webpack-react

File: client/components/SigninHeader/SigninHeaderLogo.js (hunk 6 above)
Comment (radiansz) [inline client/components/SigninHeader/SigninHeaderLogo.js]: Probably wont be here after merge of #292 in webpack-react

File: client/components/SigninHeader/SigninHeader.js (hunk 4 above)
Comment (radiansz) [inline client/components/SigninHeader/SigninHeader.js]: Probably wont be here after merge of #292 in webpack-react

File: client/components/SigninHeader/SigninHeader.js (hunk 5 above)
Comment (radiansz) [inline client/components/SigninHeader/SigninHeader.js]: Probably wont be here after merge of #292 in webpack-react

**Pull request closed without merging**